src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -81,6 +81,85 @@ private static boolean matches(CharSequence path, int start, String what, boolea
     return false;
   }
 
+  /**
+   * Normalizes a path as per <a href="http://tools.ietf.org/html/rfc3986#section-5.2.4>rfc3986</a>.
+   *
+   * There are 2 extra transformations that are not part of the spec but kept for backwards compatibility:
+   *
+   * double slash // will be converted to single slash and the path will always start with slash.
+   *
+   * @param pathname raw path
+   * @return normalized path
+   */
+  public static String normalizePath(String pathname) {
```

Comment:
I think we should actually disallow `null` here

Reply:
`null` paths will return `null` and docs are updated stating that nothing can be assumed about null so they stay null.

Reply:
ok, anyway this is internal